packages/s3-credentials-endpoint/tests/test-index.js

```diff
@@ -119,11 +128,33 @@ test('An s3credential request with expired accessToken redirects to Oauth2 provi
     .set('Accept', 'application/json')
     .set('Cookie', [`accessToken=${accessTokenRecord.accessToken}`])
     .expect(307);
+  const authorizationUrl = buildEarthdataLoginClient().getAuthorizationUrl(response.req.path);
 
   t.is(response.status, 307);
   t.is(response.headers.location, authorizationUrl);
 });
 
+test('A redirect request returns a response with an unexpired cookie ', async (t) => {
+  const {
+    stubbedAccessToken,
+  } = context;
+
+  const response = await request(distributionApp)
+    .get('/redirect')
+    .query({ code: randomId('code'), state: 'original-path/s3credentials' })
+    .set('Accept', 'application/json')
+    .expect(307);
+
+  const cookie = response.headers['set-cookie'].map(Cookie.parse);
+  const accessToken = cookie.find((c) => c.key === 'accessToken');
+  t.truthy(accessToken);
+  t.is(accessToken.value, stubbedAccessToken.accessToken);
+  t.is(
```

Comment:
I could be wrong, but I don't think this test proves that the token is unexpired. It just proves that the **token cookie value equals the return from the access token stubbed response**. If the stubbed response returned with a time that was already expired, I think this test would still pass. Feel free to let me know if I'm wrong though.

To prove that the cookie is not expired, I think we can do something like `t.true(accessToken.expires.valueOf > Date.now())`, assuming `accessToken.expires.valueOf ` is a time value in milliseconds

Ultimately, whether the cookie would be expired does depend on the access token response, but since we're in control of that response via stubbing, I think we should ensure that **if the access token response is not expired**, then the cookie is not either, since that is the bug we discovered.

Reply:
Yes, you are correct. It doesn't explicitly test that it is unexpired directly, but it does so implicitly by checking that the value of the access token matches the one created from the access token factory which would never be `Date.now()` since it is `expirationTime: moment().unix() + 60 * 60`. I can update it to be more explicit though so it is clear that we directly test for it.

Reply:
Yeah, the problem would be that without an assertion for the cookie not actually being expired, the test could have still passed if that stub was updated to produce an expired value, so the test wouldn't have been proving the intended behavior. But I see that you've added the explicit assertion for this, so we should be good.

Reply:
Ah, true, that makes sense to me